chrome/content/zotero/elements/editableText.js

```diff
@@ -185,111 +185,21 @@
-						this._ignoredWindowInactiveBlur = true;
-						return;
-					}
-					this._ignoredWindowInactiveBlur = false;
-					
-					this.dispatchEvent(new Event('blur'));
-					this.classList.remove("focused");
-					this._input.scrollLeft = 0;
-					this._input.setSelectionRange(0, 0);
-					this.removeAttribute("mousedown");
-					delete this._input.dataset.initialValue;
-				});
-				input.addEventListener('keydown', (event) => {
-					if (event.key === 'Enter') {
-						if (this.multiline === event.shiftKey) {
-							event.preventDefault();
-							this.dispatchEvent(new CustomEvent('escape_enter'));
-							this._input.blur();
-						}
-					}
-					else if (event.key === 'Escape') {
-						this.dispatchEvent(new CustomEvent('escape_enter'));
-						let initialValue = this._input.dataset.initialValue ?? '';
-						this.setAttribute('value', initialValue);
-						this._input.value = initialValue;
-						this._input.blur();
-					}
-				});
-				input.addEventListener('mousedown', (event) => {
-					// Prevent a right-click from focusing the input when unfocused
-					if (event.button === 2 && document.activeElement !== this._input) {
-						event.preventDefault();
-					}
-				});
-				input.addEventListener('dragover', (event) => {
-					// If the input is not focused, override the default drop behavior
-					if ((document.activeElement !== this._input || Services.focus.activeWindow !== window)
-							&& !this.readOnly
-							&& event.dataTransfer.getData('text/plain')) {
-						event.preventDefault();
-						event.dataTransfer.dropEffect = 'copy';
-					}
-				});
-				input.addEventListener('drop', (event) => {
-					// If the input is not focused, replace its entire value with the dropped text
-					// Otherwise, the normal drop effect takes place and the text is inserted at the cursor
-					if ((document.activeElement !== this._input || Services.focus.activeWindow !== window)
-							&& !this.readOnly
-							&& event.dataTransfer.getData('text/plain')) {
-						event.preventDefault();
-						document.activeElement?.blur();
-						// Wait a tick to work around an apparent Firefox bug where the cursor stays inside the old
-						// input even though the new input becomes visually focused
-						setTimeout(() => {
-							this.focus();
-							this._input.value = event.dataTransfer.getData('text/plain');
-							handleInput();
-						});
-					}
-				});
+				input.addEventListener('input', this._handleInput);
+				input.addEventListener('change', this._handleChange);
+				input.addEventListener('focus', this._handleFocus);
+				input.addEventListener('blur', this._handleBlur);
+				input.addEventListener('keydown', this._handleKeyDown);
+				input.addEventListener('mousedown', this._handleMouseDown);
+				input.addEventListener('dragover', this._handleDragOver);
+				input.addEventListener('drop', this._handleDrop);
 				
-				let focused = false;
+				let focused = this._input && document.activeElement === this._input;
```

Comment:
The old `focused` calculation was 100% unnecessarily convoluted. This looks identical to [the focused getter at the bottom](https://github.com/zotero/zotero/blob/f9e25675f0cc4060dc59e2471d08f1fa45cd1099/chrome/content/zotero/elements/editableText.js#L383) to me - could `this.focused` be used here instead? That way, we'd have one less variable floating around